text/0000-natvis.md

```diff
@@ -376,10 +407,77 @@ developers which may lead to a period of learning the syntax. Since this feature
 would be optional, a consumer of a crate that has natvis definitions for types
 would not need to go through this learning curve. 
 
-Not doing this would keep the existing debugging experience for external Rust crates.
-Most Rust types, outside of the standard library, do not have any debugger views
-defined for them by default which makes them difficult to interpret when viewed
-under a debugger.
+## Alternatives
+
+### Alternative 1: existing -C link-arg flag
+
+Supporting this option would mean that changes to rustc are not necessary.
+The changes would be limited to Cargo, which would be responsible for collecting
+the set of Natvis files and passing `-Clink-arg=/NATVIS:{file-path}` for each
+Natvis file.
+
+The drawbacks for this option is that it will only collect Natvis files for the
+top-most manifest. This will not walk the dependency graph and find all relevant
+Natvis files so this will only work for targets that produce a `DLL` or `EXE` and
+not an `.rlib`.
+
+### Alternative 2: custom build script to set /NATVIS linker flag
+
+Supporting this option would mean that changes to cargo and rustc are not necessary.
+Each individual crate would be able to create a custom build script that would set
+the rustc `link-arg` flag `cargo:rustc-link-arg=/NATVIS:{file-path}` for each Natvis
+file.
+
+The drawbacks for this option is that it would force all Rust developers to manually
```

Comment:
Well, those that have to add natvis files also need to do some work, the build script seems like the smallest work here. A build script is still annoying, I fully agree with that

Reply:
> the build script seems like the smallest work here. A build script is still annoying, I fully agree with that.

This is true but as with my above comment, I'm not sure the build script would be able to account for transitive dependencies. Using the `link-arg` flag would work if the current crate being built is not an `rlib` but once again that would only add Natvis solely for the current crate.